include/circt/Dialect/FIRRTL/FIRRTLTypes.h

```diff
@@ -133,6 +133,9 @@ class FIRRTLBaseType
   /// Return a 'const' or non-'const' version of this type.
   FIRRTLBaseType getConstType(bool isConst);
 
+  /// Return this type with a 'const' modifiers dropped
+  FIRRTLBaseType getAllConstDroppedType();
```

Comment:
I think this function is only used in the pass.  I have a feeling we will never use it anywhere else, so consider moving it in to the pass.

As a separate consideration, why does `type.getConstType(false)` not recursively remove const on aggregate types?  What is the use-case for that?

Reply:
As mentioned out of band, `getConstType(false)` originally recursively removed const. @darthscsi requested that it just modify the outer type on the [original const pr](https://github.com/llvm/circt/pull/4557#discussion_r1152571768).

Reply:
It turns out I was using it in `MultibitMuxOp::inferReturnType` in the next pr. But it also looks like `MultibitMuxOp` is only created during the `LowerTypes` pass (i.e. it isn't a prim op), which runs after the `DropConst`, so I shouldn't need to worry about it.